### PR TITLE
QSP-14 consistent file permissions

### PR DESCRIPTION
**Pull request by shayzluf**
**What type of PR is this?**

Bug fix

**What does this PR do? Why is it needed?**
Create files with consistent file permissions for DB,Backup,Log

**Which issues(s) does this PR fix?**

Part of #6327

### CODE REVIEW

File: beacon-chain/db/kv/backup.go [hunk 1]
```diff
@@ -35,7 +35,7 @@ func (kv *Store) Backup(ctx context.Context) error {
 	backupPath := path.Join(backupsDir, fmt.Sprintf("prysm_beacondb_at_slot_%07d.backup", head.Block.Slot))
 	logrus.WithField("prefix", "db").WithField("backup", backupPath).Info("Writing backup database.")
 
-	copyDB, err := bolt.Open(backupPath, 0666, nil)
+	copyDB, err := bolt.Open(backupPath, 0600, nil)
```
Comment (terencechain) [inline beacon-chain/db/kv/backup.go:38]: Should we define `0600` as a config and use it through out?

Comment (shayzluf) [reply beacon-chain/db/kv/backup.go:38]: sg

File: shared/params/BUILD.bazel [hunk 2]
```diff
@@ -4,6 +4,7 @@ load("@io_bazel_rules_go//go:def.bzl", "go_test")
 go_library(
     name = "go_default_library",
     srcs = [
+        "IoConfig.go",
```
Comment (rauljordan) [inline shared/params/BUILD.bazel:7]:
```suggestion
        "io_config.go",
```

File: shared/params/io_config.go [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+import "os"
+
+// IoConfig defines the shared io parameters.
+type IoConfig struct {
+	FilePermission os.FileMode
+}
+
+var defaultIoConfig = &IoConfig{
+	FilePermission: 0600, //-rw------- Read and Write permissions for user
```
Comment (rauljordan) [inline shared/params/io_config.go:11]: Can we name this to be more clear? Something like `ReadWritePermissions`

Comment (shayzluf) [reply shared/params/io_config.go]: renamed

**Pull request merged**